web/react/utils/utils.jsx

```diff
@@ -884,6 +884,23 @@ export function getDisplayName(user) {
     return user.username;
 }
 
+export function displayUsername(userId) {
+    const user = UserStore.getProfile(userId);
+    const nameFormat = PreferenceStore.getPreference(Constants.Preferences.CATEGORY_DISPLAY_SETTINGS, 'name_format', {value: 'false'}).value;
+
+    let username;
+    if (nameFormat === 'nickname_full_name') {
+        username = user.nickname || getFullName();
+    } else if (nameFormat === 'full_name') {
+        username = getFullName();
+    }
+    if (!username.trim().length) {
```

Comment:
This will throw a reference error if nameFormat is username since username will be undefined at that point